cdap-elastic/src/test/java/io/cdap/cdap/metadata/elastic/ElasticsearchMetadataStorageTest.java

```diff
@@ -380,4 +380,44 @@ protected void validateCursor(String cursor, int expectedOffset, int expectedPag
     Assert.assertEquals(expectedOffset, c.getOffset());
     Assert.assertEquals(expectedPageSize, c.getLimit());
   }
+
+  @Test
+  public void testSearchReturnsRequiredTerms() throws IOException {
+    MetadataStorage mds = getMetadataStorage();
+    MutationOptions options = MutationOptions.builder().setAsynchronous(false).build();
+
+    List<MetadataRecord> records = IntStream.range(0, 3).boxed().map(i -> new MetadataRecord(
+        MetadataEntity.ofDataset("ns" + i, "ds" + i),
+        new Metadata(MetadataScope.USER, tags("tag", "t" + i), props("p", "v" + i)))).collect(Collectors.toList());
+    mds.batch(records.stream().map(r -> new Update(r.getEntity(), r.getMetadata())).collect(Collectors.toList()),
+        options);
+
+    // does it return a single required dataset?
+    SearchRequest request = SearchRequest.of("+t1").build();
+    SearchResponse response = mds.search(request);
+    Assert.assertEquals(1, response.getResults().size());
```

Comment:
I think you should not only test the number of results returned, but make sure it is the right set of entities.

Reply:
Addressed in the latest commit.

Reply:
Can you also test that the entity is the expected entity?

Reply:
Each record consists of a metadata entity and the metadata for the entity. Here you only test that the metadata is correct, bit you do not validate that the metadata entity is the correct one, in this case, `MetadataEntity.ofDataset("ns1", "ds1")`.

Reply:
I see. In that case, would it be most concise to simply check the equality of the two objects?

Reply:
yes

Reply:
Addressed in latest commit.